### DISCUSSION TITLE
How to use matrix and boolean?

### DISCUSSION THREAD

**Discussion by BotellaA** [community/community#25555] [Actions]
<p>Hello,</p>
<p>I cannot find a way to make a boolean in a matrix to work… What am I missing?</p>
<p>Thanks for you help</p>
<pre><code>jobs:
  build-ubuntu:
    runs-on: ubuntu-18.04
    needs: format
    strategy:
      matrix:
        config:
        - {cc: gcc, cxx: g++, cmake_args: "-DCMAKE_BUILD_TYPE=Debug", coverage: 'yes', benchmark: 'no'}
        - {cc: gcc, cxx: g++, cmake_args: "-DCMAKE_BUILD_TYPE=Release -DUSE_BENCHMARK=ON", coverage: 'no', benchmark: 'yes'}
        - {cc: clang, cxx: clang++, cmake_args: "-DCMAKE_BUILD_TYPE=Debug", coverage: 'no', benchmark: 'no'}
        - {cc: clang, cxx: clang++, cmake_args: "-DCMAKE_BUILD_TYPE=Release", coverage: 'no', benchmark: 'no'}

    steps:
    - uses: actions/checkout@v1
    - name: Compile
      run: |
        mkdir -p build
        cd build
        cmake ${{ matrix.config.cmake_args }} ..
        cmake --build . -- -j2
      env:
        CC: ${{ matrix.config.cc }}
        CXX: ${{ matrix.config.cxx }}
    - name: Test
      run: |
        mkdir -p test
        cd test
        cmake ${{ matrix.config.cmake_args }} -DCMAKE_PREFIX_PATH:PATH=$GITHUB_WORKSPACE/build/opengeode/install ../tests
        cmake --build . -- -j2
        ctest --output-on-failure
      env:
        CC: ${{ matrix.config.cc }}
        CXX: ${{ matrix.config.cxx }}
    - name: Kcov
      run: curl -s https://raw.githubusercontent.com/Geode-solutions/actions/master/kcov.sh | bash -s -- ${GITHUB_WORKSPACE}/build/opengeode/bin
      if: matrix.config.coverage == 'yes' &amp;&amp; (github.event_name == 'pull_request' || github.ref == 'refs/heads/master')
</code></pre>

**Comment by chrispat** [comment, marked as answer]
<p>This works for me.</p>
<pre><code>name: Boolean Matrix

on:
  push:
    branches:
      - boolean-matrix

jobs:
  build:

    runs-on: ubuntu-latest
    strategy:
      matrix:
        config:
        - {run-it: true}
        - {run-it: false}

    steps:
    - name: Run a one-line script
      run: echo Hello, world!
      if: matrix.config.run-it
</code></pre>
<p><a href="https://github.com/chrispat/test-action/runs/319324546" rel="nofollow noopener">https://github.com/chrispat/test-action/runs/319324546</a></p>
<p>However, given you are doing string comparison below I would also expect that to work. </p>
